==25969== Helgrind, a thread error detector
==25969== Copyright (C) 2007-2017, and GNU GPL'd, by OpenWorks LLP et al.
==25969== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==25969== Command: ./bin/kernel ESTABILIDAD_GENERAL 0 estabilidad_general.config
==25969== 
==25969== ---Thread-Announcement------------------------------------------
==25969== 
==25969== Thread #3 was created
==25969==    at 0x49A19F3: clone (clone.S:76)
==25969==    by 0x49A28EE: __clone_internal (clone-internal.c:83)
==25969==    by 0x49106D8: create_thread (pthread_create.c:295)
==25969==    by 0x49111FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==25969==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x10E6F8: inicializar_servidores (kernel.c:123)
==25969==    by 0x10E096: main (kernel.c:7)
==25969== 
==25969== ---Thread-Announcement------------------------------------------
==25969== 
==25969== Thread #1 is the program's root thread
==25969== 
==25969== ----------------------------------------------------------------
==25969== 
==25969== Possible data race during write of size 1 at 0x4AA8A50 by thread #3
==25969== Locks held: none
==25969==    at 0x48597B4: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4907664: _IO_new_file_xsputn (fileops.c:1235)
==25969==    by 0x4907664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48F300B: outstring_func (vfprintf-internal.c:239)
==25969==    by 0x48F300B: __vfprintf_internal (vfprintf-internal.c:1517)
==25969==    by 0x48DC79E: printf (printf.c:33)
==25969==    by 0x4874A34: txt_write_in_stdout (txt.c:32)
==25969==    by 0x48757EF: _log_write_in_level (log.c:166)
==25969==    by 0x487537E: log_debug (log.c:95)
==25969==    by 0x1107D6: esperar_cliente (socket.c:56)
==25969==    by 0x10E7BD: iniciar_cpu_dispatch (kernel.c:139)
==25969==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4910AC2: start_thread (pthread_create.c:442)
==25969==    by 0x49A1A03: clone (clone.S:100)
==25969== 
==25969== This conflicts with a previous write of size 1 by thread #1
==25969== Locks held: none
==25969==    at 0x4859796: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4907664: _IO_new_file_xsputn (fileops.c:1235)
==25969==    by 0x4907664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48FCF1B: puts (ioputs.c:40)
==25969==    by 0x10E0A5: main (kernel.c:9)
==25969==  Address 0x4aa8a50 is 0 bytes inside a block of size 1,024 alloc'd
==25969==    at 0x484A919: malloc (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x48FABA3: _IO_file_doallocate (filedoalloc.c:101)
==25969==    by 0x4909CDF: _IO_doallocbuf (genops.c:347)
==25969==    by 0x4908F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:744)
==25969==    by 0x49076D4: _IO_new_file_xsputn (fileops.c:1243)
==25969==    by 0x49076D4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48F300B: outstring_func (vfprintf-internal.c:239)
==25969==    by 0x48F300B: __vfprintf_internal (vfprintf-internal.c:1517)
==25969==    by 0x48DC79E: printf (printf.c:33)
==25969==    by 0x4874A34: txt_write_in_stdout (txt.c:32)
==25969==    by 0x48757EF: _log_write_in_level (log.c:166)
==25969==    by 0x487537E: log_debug (log.c:95)
==25969==    by 0x10E3C1: inicializar_kernel (kernel.c:64)
==25969==    by 0x10E08C: main (kernel.c:6)
==25969==  Block was alloc'd by thread #1
==25969== 
==25969== ---Thread-Announcement------------------------------------------
==25969== 
==25969== Thread #4 was created
==25969==    at 0x49A19F3: clone (clone.S:76)
==25969==    by 0x49A28EE: __clone_internal (clone-internal.c:83)
==25969==    by 0x49106D8: create_thread (pthread_create.c:295)
==25969==    by 0x49111FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==25969==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x10E715: inicializar_servidores (kernel.c:124)
==25969==    by 0x10E096: main (kernel.c:7)
==25969== 
==25969== ----------------------------------------------------------------
==25969== 
==25969== Possible data race during write of size 1 at 0x4AAA8E7 by thread #4
==25969== Locks held: none
==25969==    at 0x4859796: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4907664: _IO_new_file_xsputn (fileops.c:1235)
==25969==    by 0x4907664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48FBA9C: fputs (iofputs.c:38)
==25969==    by 0x48749FA: txt_write_in_file (txt.c:27)
==25969==    by 0x48757A1: _log_write_in_level (log.c:158)
==25969==    by 0x487537E: log_debug (log.c:95)
==25969==    by 0x1107D6: esperar_cliente (socket.c:56)
==25969==    by 0x10FBA6: iniciar_cpu_interrupt (kernel.c:549)
==25969==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4910AC2: start_thread (pthread_create.c:442)
==25969==    by 0x49A1A03: clone (clone.S:100)
==25969== 
==25969== This conflicts with a previous write of size 1 by thread #3
==25969== Locks held: none
==25969==    at 0x4859796: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4907664: _IO_new_file_xsputn (fileops.c:1235)
==25969==    by 0x4907664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48FBA9C: fputs (iofputs.c:38)
==25969==    by 0x48749FA: txt_write_in_file (txt.c:27)
==25969==    by 0x48757A1: _log_write_in_level (log.c:158)
==25969==    by 0x487537E: log_debug (log.c:95)
==25969==    by 0x1107D6: esperar_cliente (socket.c:56)
==25969==    by 0x10E7BD: iniciar_cpu_dispatch (kernel.c:139)
==25969==  Address 0x4aaa8e7 is 71 bytes inside a block of size 4,096 alloc'd
==25969==    at 0x484A919: malloc (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x48FABA3: _IO_file_doallocate (filedoalloc.c:101)
==25969==    by 0x4909CDF: _IO_doallocbuf (genops.c:347)
==25969==    by 0x4908F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:744)
==25969==    by 0x49076D4: _IO_new_file_xsputn (fileops.c:1243)
==25969==    by 0x49076D4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48FBA9C: fputs (iofputs.c:38)
==25969==    by 0x48749FA: txt_write_in_file (txt.c:27)
==25969==    by 0x48757A1: _log_write_in_level (log.c:158)
==25969==    by 0x487537E: log_debug (log.c:95)
==25969==    by 0x10E3C1: inicializar_kernel (kernel.c:64)
==25969==    by 0x10E08C: main (kernel.c:6)
==25969==  Block was alloc'd by thread #1
==25969== 
==25969== ---Thread-Announcement------------------------------------------
==25969== 
==25969== Thread #6 was created
==25969==    at 0x49A19F3: clone (clone.S:76)
==25969==    by 0x49A28EE: __clone_internal (clone-internal.c:83)
==25969==    by 0x49106D8: create_thread (pthread_create.c:295)
==25969==    by 0x49111FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==25969==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x10FBF4: iniciar_cpu_interrupt (kernel.c:555)
==25969==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4910AC2: start_thread (pthread_create.c:442)
==25969==    by 0x49A1A03: clone (clone.S:100)
==25969== 
==25969== ---Thread-Announcement------------------------------------------
==25969== 
==25969== Thread #5 was created
==25969==    at 0x49A19F3: clone (clone.S:76)
==25969==    by 0x49A28EE: __clone_internal (clone-internal.c:83)
==25969==    by 0x49106D8: create_thread (pthread_create.c:295)
==25969==    by 0x49111FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==25969==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x10E80B: iniciar_cpu_dispatch (kernel.c:145)
==25969==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4910AC2: start_thread (pthread_create.c:442)
==25969==    by 0x49A1A03: clone (clone.S:100)
==25969== 
==25969== ----------------------------------------------------------------
==25969== 
==25969== Possible data race during write of size 1 at 0x4AAA8A0 by thread #6
==25969== Locks held: none
==25969==    at 0x48597B4: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4907664: _IO_new_file_xsputn (fileops.c:1235)
==25969==    by 0x4907664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48FBA9C: fputs (iofputs.c:38)
==25969==    by 0x48749FA: txt_write_in_file (txt.c:27)
==25969==    by 0x48757A1: _log_write_in_level (log.c:158)
==25969==    by 0x487537E: log_debug (log.c:95)
==25969==    by 0x111258: saludar_cliente_generico (socket.c:312)
==25969==    by 0x10FC7F: atender_interrupt (kernel.c:565)
==25969==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4910AC2: start_thread (pthread_create.c:442)
==25969==    by 0x49A1A03: clone (clone.S:100)
==25969== 
==25969== This conflicts with a previous write of size 1 by thread #5
==25969== Locks held: none
==25969==    at 0x4859796: mempcpy (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4907664: _IO_new_file_xsputn (fileops.c:1235)
==25969==    by 0x4907664: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48FBA9C: fputs (iofputs.c:38)
==25969==    by 0x48749FA: txt_write_in_file (txt.c:27)
==25969==    by 0x48757A1: _log_write_in_level (log.c:158)
==25969==    by 0x487537E: log_debug (log.c:95)
==25969==    by 0x111258: saludar_cliente_generico (socket.c:312)
==25969==    by 0x10E878: atender_dispatch (kernel.c:154)
==25969==  Address 0x4aaa8a0 is 0 bytes inside a block of size 4,096 alloc'd
==25969==    at 0x484A919: malloc (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x48FABA3: _IO_file_doallocate (filedoalloc.c:101)
==25969==    by 0x4909CDF: _IO_doallocbuf (genops.c:347)
==25969==    by 0x4908F5F: _IO_file_overflow@@GLIBC_2.2.5 (fileops.c:744)
==25969==    by 0x49076D4: _IO_new_file_xsputn (fileops.c:1243)
==25969==    by 0x49076D4: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1196)
==25969==    by 0x48FBA9C: fputs (iofputs.c:38)
==25969==    by 0x48749FA: txt_write_in_file (txt.c:27)
==25969==    by 0x48757A1: _log_write_in_level (log.c:158)
==25969==    by 0x487537E: log_debug (log.c:95)
==25969==    by 0x10E3C1: inicializar_kernel (kernel.c:64)
==25969==    by 0x10E08C: main (kernel.c:6)
==25969==  Block was alloc'd by thread #1
==25969== 
==25969== 
==25969== Process terminating with default action of signal 2 (SIGINT)
==25969==    at 0x490D117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==25969==    by 0x490D117: __futex_abstimed_wait_common (futex-internal.c:87)
==25969==    by 0x490D117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==25969==    by 0x4918C77: __new_sem_wait_slow64.constprop.0 (sem_waitcommon.c:183)
==25969==    by 0x48540A8: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x10CA36: wait_sem (semaforos.c:55)
==25969==    by 0x10E10B: main (kernel.c:18)
==25969== ---Thread-Announcement------------------------------------------
==25969== 
==25969== Thread #27 was created
==25969==    at 0x49A19F3: clone (clone.S:76)
==25969==    by 0x49A28EE: __clone_internal (clone-internal.c:83)
==25969==    by 0x49106D8: create_thread (pthread_create.c:295)
==25969==    by 0x49111FF: pthread_create@@GLIBC_2.34 (pthread_create.c:828)
==25969==    by 0x4853767: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x10E64E: inicializar_planificacion (kernel.c:107)
==25969==    by 0x10E0FC: main (kernel.c:15)
==25969== 
==25969== ----------------------------------------------------------------
==25969== 
==25969== Thread #27: Exiting thread still holds 1 lock
==25969==    at 0x490D117: __futex_abstimed_wait_common64 (futex-internal.c:57)
==25969==    by 0x490D117: __futex_abstimed_wait_common (futex-internal.c:87)
==25969==    by 0x490D117: __futex_abstimed_wait_cancelable64 (futex-internal.c:139)
==25969==    by 0x4918C77: __new_sem_wait_slow64.constprop.0 (sem_waitcommon.c:183)
==25969==    by 0x48540A8: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x10CA36: wait_sem (semaforos.c:55)
==25969==    by 0x10BAD1: planificar_largo_plazo_PMCP (scheduler.c:410)
==25969==    by 0x485396A: ??? (in /usr/libexec/valgrind/vgpreload_helgrind-amd64-linux.so)
==25969==    by 0x4910AC2: start_thread (pthread_create.c:442)
==25969==    by 0x49A1A03: clone (clone.S:100)
==25969== 
==25969== 
==25969== Use --history-level=approx or =none to gain increased speed, at
==25969== the cost of reduced accuracy of conflicting-access information
==25969== For lists of detected and suppressed errors, rerun with: -s
==25969== ERROR SUMMARY: 13024 errors from 4 contexts (suppressed: 3801 from 252)
